war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -1071,36 +1084,36 @@ function helpButtonOnClick() {
   if (div.style.display != "block") {
     div.style.display = "block";
     // make it visible
-    new Ajax.Request(this.getAttribute("helpURL"), {
-      method: "get",
-      onSuccess: function (x) {
-        // Which plugin is this from?
-        var from = x.getResponseHeader("X-Plugin-From");
-        div.innerHTML =
-          x.responseText +
-          (from ? "<div class='from-plugin'>" + from + "</div>" : "");
-
-        // Ensure links open in new window unless explicitly specified otherwise
-        var links = div.getElementsByTagName("a");
-        for (var i = 0; i < links.length; i++) {
-          var link = links[i];
-          if (link.hasAttribute("href")) {
-            // ignore document anchors
-            if (!link.hasAttribute("target")) {
-              link.setAttribute("target", "_blank");
-            }
-            if (!link.hasAttribute("rel")) {
-              link.setAttribute("rel", "noopener noreferrer");
+
+    fetch(this.getAttribute("helpURL")).then((rsp) => {
+      rsp.text().then((responseText) => {
+        if (rsp.ok) {
+          var from = rsp.headers.get("X-Plugin-From");
+          // Which plugin is this from?
+          div.innerHTML =
+            responseText +
+            (from ? "<div class='from-plugin'>" + from + "</div>" : "");
+
+          // Ensure links open in new window unless explicitly specified otherwise
+          var links = div.getElementsByTagName("a");
+          for (var i = 0; i < links.length; i++) {
+            var link = links[i];
+            if (link.hasAttribute("href")) {
+              // ignore document anchors
+              if (!link.hasAttribute("target")) {
+                link.setAttribute("target", "_blank");
+              }
+              if (!link.hasAttribute("rel")) {
+                link.setAttribute("rel", "noopener noreferrer");
+              }
             }
           }
+        } else {
+          div.innerHTML =
+            "<b>ERROR</b>: Failed to load help file: " + responseText;
```

Comment:
The original was printing the HTTP status text sent by the server, not the text body of the response.

```suggestion
            "<b>ERROR</b>: Failed to load help file: " + rsp.statusText;
```